openedx/core/lib/blockstore_api/methods.py

```diff
@@ -35,16 +36,26 @@
 from .config import use_blockstore_app
 
 
+log = logging.getLogger(__name__)
+
+
 def toggle_blockstore_api(func):
     """
     Decorator function to toggle usage of the Blockstore service
     and the in-built Blockstore app dependency.
     """
     @wraps(func)
     def wrapper(*args, **kwargs):
+        joined_args = " "
+        log.info('blockstore %s API call called with arguments %s', func.__name__, joined_args.join(str(args)))
```

Comment:
This results in log messages like this:

> INFO 845 [openedx.core.lib.blockstore_api.methods] [user 8] [ip 172.21.0.18] methods.py:50 - blockstore get_bundle_version_files API call called with arguments ( U U I D ( ' e c e 1 6 b 7 0 - 4 d 1 0 - 4 1 9 3 - a 6 4 7 - 9 d b 8 5 4 a 1 c 4 7 0 ' ) ,   2 1 2 )

You don't need all that `joined_args` business, the `args` list will just stringify straight like this (and I've also added the `kwargs` dict, since some invocations may use them):

```suggestion
        log.info('blockstore %s API call called with arguments %s, kwargs=%s', func.__name__, args, kwargs)
```

Results in logs like this:

> INFO 946 [openedx.core.lib.blockstore_api.methods] [user 8] [ip 172.21.0.18] methods.py:50 - blockstore get_bundle API call called with arguments (UUID('3a9293d7-9846-4de4-819e-0ac35e9d996b'),), kwargs={}

Reply:
I agree that the format you suggest is better -- thanks. But am concerned by the lack of timestamps in the logs you pasted. Did you edit them out, or were they missing altogether? Without the timestamps, these logs won't fulfill the purpose they're intended for.

Reply:
Oh sorry, I clipped the prefix off these logs for brevity's sake. There are timestamps, `log.info` takes care of that:

> edx.devstack.lms                  | 2022-08-30 05:49:56,424 INFO 2199 [openedx.core.lib.blockstore_api.methods] [user 8] [ip 172.21.0.18] methods.py:50 - blockstore get_bundle API call called with arguments (UUID('c895fe42-48ef-4525-a486-580ba65e566e'),), kwargs={}
> edx.devstack.lms                  | 2022-08-30 05:49:56,500 INFO 2199 [openedx.core.lib.blockstore_api.methods] [user 8] [ip 172.21.0.18] methods.py:57 - IDA blockstore get_bundle API call is done